packages/expo-camera/ios/EXCamera/EXCamera.m

```diff
@@ -540,7 +540,33 @@ - (void)record:(NSDictionary *)options resolve:(UMPromiseResolveBlock)resolve re
       [connection setPreferredVideoStabilizationMode:self.videoStabilizationMode];
     }
     [connection setVideoOrientation:[EXCameraUtils videoOrientationForDeviceOrientation:[[UIDevice currentDevice] orientation]]];
-
+    
+    if (options[@"codec"]) {
+        AVVideoCodecType videoCodecType = [EXCameraUtils videoCodecForType: [options[@"codec"] integerValue]];
+        
+        if ([self.movieFileOutput.availableVideoCodecTypes containsObject: videoCodecType]) {
+            [self.movieFileOutput setOutputSettings: @{AVVideoCodecKey: videoCodecType} forConnection: connection];
+            
+            self.videoCodecType = videoCodecType;
+        }
+        
+        else {
+            UMLogWarn(@"%s: Video Codec '%@' is not supported on this device.", __func__, videoCodecType);
+            
+            NSString *videoCodecErrorMessage = [NSString stringWithFormat: @"Video Codec '%@' is not supported on this device", videoCodecType];
+            
+            reject(@"E_RECORDING_FAILED", videoCodecErrorMessage, nil);
+            
+            [self cleanupMovieFileCapture];
+            _videoRecordedResolve = nil;
+            _videoRecordedReject = nil;
+            
+            return;
+        }
+    }
+     
+    
+    
```

Comment:
```suggestion
    if (options[@"codec"]) {
      AVVideoCodecType videoCodecType = [EXCameraUtils videoCodecForType: [options[@"codec"] integerValue]];

      if ([self.movieFileOutput.availableVideoCodecTypes containsObject:videoCodecType]) {
        [self.movieFileOutput setOutputSettings: @{AVVideoCodecKey: videoCodecType} forConnection: connection];    
        self.videoCodecType = videoCodecType;
      } else {
        NSString *videoCodecErrorMessage = [NSString stringWithFormat: @"Video Codec '%@' is not supported on this device", videoCodecType];
        reject(@"E_RECORDING_FAILED", videoCodecErrorMessage, nil);

        [self cleanupMovieFileCapture];
        _videoRecordedResolve = nil;
        _videoRecordedReject = nil;

        return;
      }
    }

```
We don't want some extra `warns` that pollutes the user-space. Rejecting with proper, descriptive error message is sufficient 😉